splunk_add_on_ucc_framework/tabs/proxy_tab.py

```diff
@@ -0,0 +1,208 @@
+            "minLength": 0,
+            "maxLength": 8192,
+        }
+    ],
+    "encrypted": True,
+    "field": "proxy_password",
+}
+
+DEFAULT_DNS_RESOLUTION = {
+    "type": "checkbox",
+    "label": "DNS resolution",
+    "field": "proxy_rdns",
+}
+
+
+class ProxyTab(Tab):
+    @property
+    def tab_type(self) -> Optional[str]:
+        return "proxyTab"
+
+    @classmethod
+    def from_definition(cls, definition: Dict[str, Any]) -> Optional["Tab"]:
+        """
+        This function checks if the definition has type == ProxyTab; if it does, it gets converted;
+        otherwise, it returns None.
+        """
+        if definition.get("type") == "proxyTab":
+            entity = []
+            entity_key_const_dict = {
+                "enable_proxy": DEFAULT_PROXY_ENABLE,
+                "proxy_type": DEFAULT_PROXY_TYPE,
+                "host": DEFAULT_HOST,
+                "port": DEFAULT_PORT,
+                "username": DEFAULT_USERNAME,
+                "password": DEFAULT_PASSWORD,
+                "dns_resolution": DEFAULT_DNS_RESOLUTION,
+            }
+
+            def updating_dictionaries(key_name: str, const: Any) -> None:
+                for key, value in const.items():
+                    nonlocal definition
+                    if key not in definition.get(key_name, {}):
+                        definition[key_name][key] = value
+                entity.append(definition.get(key_name))
+
+            if ("username" not in definition) ^ ("password" not in definition):
+                logger.error("Either of username or password is not mentioned.")
+                sys.exit(1)
+
+            elif definition.get("username") != definition.get("password"):
+                if (
+                    isinstance(definition.get("username"), dict)
+                    and definition.get("password") is False
+                ):
+                    logger.error(
+                        "You had updated the username but set the password to 'false' which is not allowed "
+                        "set `password = True` for default configuration."
+                    )
+                    sys.exit(1)
+                elif (
+                    isinstance(definition.get("password"), dict)
+                    and definition.get("username") is False
+                ):
+                    logger.error(
+                        "You had updated the password but set username to `false` which is not allowed "
+                        "set `username = True` for default configuration."
+                    )
+                    sys.exit(1)
+                elif (
+                    type(definition.get("username")) is bool
+                    and type(definition.get("password")) is bool
+                ):
+                    logger.error(
+                        "You had set conflicting values for username and password."
+                    )
+                    sys.exit(1)
```

Comment:
I don't get it, what does it mean "conflicting values"?

Reply:
Maybe we can specify exactly this?
```
You had set different values for username (True) and password (False). They should be the same

Reply:
Updated the message.